src/main/java/agh/ics/oop/SimulationEngine.java

```diff
@@ -15,6 +15,10 @@ public SimulationEngine(MoveDirection[] directions, IWorldMap map, Vector2d[] st
         this.animalList = new ArrayList<>();
         for (Vector2d position : starting_positions) {
             Animal new_animal = new Animal(map, position);
+            // Nie jestem pewien czy to jest poprawne rozwiązanie, ponieważ nie mówiliśmy o takim castowaniu,
+            // ale przynajmniej przy mojej implementacji zadań jest to (chyba) jedyne możliwe rozwiązanie...
+            // I działa więc chyba nie trzeba się przejmować ¯\_(ツ)_/¯
+            new_animal.addObserver((IPositionChangeObserver) map);
```

Comment:
Jeśli każda mapa rozszerzałaby ten interfejs to nie musielibyśmy rzutować -> interfejsy też mogą rozszerzać inne interfejsy.